### PR TITLE
MM-40838: new & reworked templates

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Introduce an `Employee Onboarding` playbook based on https://handbook.mattermost.com/operations/workplace/people/working-at-mattermost/onboarding/manager-onboarding-1#onboarding, and a bug bash playbook based on our own [Playbooks: Bug Bash](https://community.mattermost.com/playbooks/playbooks/zp1xp684zig5pg9np4ccixudoh) playbook:

CleanShot 2022-01-25 at 17 21 14@2x

CleanShot 2022-01-25 at 17 21 46@2x

CleanShot 2022-01-25 at 17 22 06@2x

As per @calebroseland, introduce [`mtrim`](https://github.com/md-vanilla/js-trim-multiline-string) for easier handling of multi-line, indented strings.

Tidy up some of the existing templates, adding descriptions and unifying styling.

Note that I'm deferring the use of "new" icons in the template selector until the pending rework of this widget.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-41090

#### Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/backstage/template_selector.tsx [hunk 1]
```diff
@@ -376,49 +410,317 @@ export const PresetTemplates: PresetTemplate[] = [
...
+                            'Complete the onboarding template prior to your new staff member\'s start date',
+                            mtrim`Managers play a large role in setting their new direct report up for success and making them feel welcome by setting clear expectations and preparing the team and internal stakeholders for how they can help new colleagues integrate and connect organizationally and culturally.
+                                * **Onboarding Objectives:** Clarify the areas and projects your new team member should focus on in their first 90 days. Use the _Overview of the Role_ that you completed when you opened the role.
+                                * **AOR clarity:** Identify AORs that are relevant for your new hire, and indicate any AORs that your new hire will [DRI](https://handbook.mattermost.com/company/about-mattermost/list-of-terms#dri) or act as backup DRI. As needed, clarify AOR transitions with internal stakholders ahead of your new hire's start date. See [AOR page](https://handbook.mattermost.com/operations/operations/areas-of-responsibility) Include the interview panel and their respective focus areas.
```
Comment (calebroseland) [inline webapp/src/components/backstage/template_selector.tsx:479]:
```suggestion
                                * **AOR clarity:** Identify AORs that are relevant for your new hire, and indicate any AORs that your new hire will [DRI](https://handbook.mattermost.com/company/about-mattermost/list-of-terms#dri) or act as backup DRI. As needed, clarify AOR transitions with internal stakeholders ahead of your new hire's start date. See [AOR page](https://handbook.mattermost.com/operations/operations/areas-of-responsibility) Include the interview panel and their respective focus areas.
```

File: webapp/src/components/backstage/template_selector.tsx [hunk 2]
```diff
@@ -376,49 +410,317 @@ export const PresetTemplates: PresetTemplate[] = [
...
+
+                                Be sure to include the hashtag \#newcolleague when posting your message.`,
+                        ),
+                        newChecklistItem(
+                            'Review Team [AORs](https://handbook.mattermost.com/operations/operations/areas-of-responsibility)',
+                            'This is also a good time to review the new hire\'s AOR and onboarding expectations.'
+                        ),
+                        newChecklistItem(
+                            'Review list of key internal partners',
+                            'Theser are individuals the new staff member will work with and who the new staff member should set up meetings with during their first month or two.',
```
Comment (calebroseland) [inline webapp/src/components/backstage/template_selector.tsx:499]:
```suggestion
                            'These are individuals the new staff member will work with and who the new staff member should set up meetings with during their first month or two.',
```

File: webapp/src/components/backstage/template_selector.tsx [hunk 3]
```diff
@@ -376,49 +410,317 @@ export const PresetTemplates: PresetTemplate[] = [
...
+                            'Review GitHub commit diff',
+                        ),
+                        newChecklistItem(
+                            'Identify new features to add to target testing areas checklist',
+                        ),
+                        newChecklistItem(
+                            'Identify existing functionality to add to target testing areas checklist',
+                        ),
+                        newChecklistItem(
+                            'Add relvant T0/E0/E10/E20 permutations',
```
Comment (calebroseland) [inline webapp/src/components/backstage/template_selector.tsx:646]:
```suggestion
                            'Add relevant T0/E0/E10/E20 permutations',
```

File: webapp/src/components/backstage/template_selector.tsx [hunk 4]
```diff
@@ -25,12 +26,35 @@ export interface PresetTemplate {
...
 
-export const PresetTemplates: PresetTemplate[] = [
+const preprocessTemplates = (presetTemplates: PresetTemplate[]): PresetTemplate[] => (
+    presetTemplates.map((pt) => ({
+        ...pt,
+        template: {
+            ...pt.template,
+            num_stages: pt.template?.checklists.length,
+            num_actions:
+				1 + // Channel creation is hard-coded
```
Comment (calebroseland) [inline webapp/src/components/backstage/template_selector.tsx:36]: Odd that editorconfig didn't pick up on that! (tabs-spaces)
```suggestion
                1 + // Channel creation is hard-coded
```

**Review by calebroseland — CHANGES_REQUESTED** [review]
LGTM! Just a nit and a few typos while we're in here (shoutouts to https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker and https://marketplace.visualstudio.com/items?itemName=ban.spellright 😛).

**Review by calebroseland — APPROVED** [review]

**Review by abhijit-singh — APPROVED** [review]

**Pull request merged**